azurerm/resource_arm_data_factory_linked_service_sql_server.go

```diff
@@ -0,0 +1,357 @@
+	client := meta.(*ArmClient).dataFactoryLinkedServiceClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	dataFactoryName := id.Path["factories"]
+	name := id.Path["linkedservices"]
+
+	response, err := client.Delete(ctx, resourceGroup, dataFactoryName, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(response) {
+			return fmt.Errorf("Error deleting Data Factory Linked Service SQL Server %q (Data Factory %q / Resource Group %q): %+v", name, dataFactoryName, resourceGroup, err)
+		}
+	}
+
+	return nil
+}
+
+// Because the password isn't returned from the api in the connection string, we'll check all
+// but the password string and return true if they match.
+func azureRmDataFactoryLinkedServiceConnectionStringDiff(k, old string, new string, d *schema.ResourceData) bool {
+	oldSplit := strings.Split(strings.ToLower(old), ";")
+	newSplit := strings.Split(strings.ToLower(new), ";")
+
+	sort.Strings(oldSplit)
+	sort.Strings(newSplit)
+
+	// We need to remove the password from the new string since it isn't returned from the api
+	for i, v := range newSplit {
+		if strings.HasPrefix(v, "password") {
+			newSplit = append(newSplit[:i], newSplit[i+1:]...)
+		}
+	}
+
+	if len(oldSplit) != len(newSplit) {
+		return false
+	}
+
+	// We'll error out if we find any differences between the old and the new connectiong strings
+	for i := range oldSplit {
+		if !strings.EqualFold(oldSplit[i], newSplit[i]) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func validateAzureRMDataFactoryLinkedServiceName(v interface{}, k string) (warnings []string, errors []error) {
+	value := v.(string)
+	if regexp.MustCompile(`^[-.+?/<>*%&:\\]+$`).MatchString(value) {
+		errors = append(errors, fmt.Errorf("any of '-' '.', '+', '?', '/', '<', '>', '*', '%%', '&', ':', '\\', are not allowed in %q: %q", k, value))
+	}
+
+	return warnings, errors
+}
+
+func expandDataFactoryLinkedServiceIntegrationRuntime(d *schema.ResourceData) *datafactory.IntegrationRuntimeReference {
+	if v, ok := d.GetOk("integration_runtime_name"); ok {
+		integrationRuntimeName := v.(string)
+		typeString := "IntegrationRuntimeReference"
+
+		return &datafactory.IntegrationRuntimeReference{
+			ReferenceName: &integrationRuntimeName,
+			Type:          &typeString,
+		}
+	}
+	return nil
+}
+
+func expandDataFactoryLinkedServiceParameters(d *schema.ResourceData) map[string]*datafactory.ParameterSpecification {
+	input := d.Get("parameters").(map[string]interface{})
+
+	output := make(map[string]*datafactory.ParameterSpecification)
+
+	for k, v := range input {
+		output[k] = &datafactory.ParameterSpecification{
+			Type:         datafactory.ParameterTypeString,
+			DefaultValue: v.(string),
+		}
+	}
+
+	return output
+}
+
+func expandDataFactoryLinkedServiceAdditionalProperties(d *schema.ResourceData) map[string]interface{} {
+	input := d.Get("additional_properties").(map[string]interface{})
```

Comment:
(as above) we could parse this prior to this method and make this:

```suggestion
func expandDataFactoryLinkedServiceAdditionalProperties(input map[string]interface{}) map[string]interface{} {
```